labrad/node/__init__.py

```diff
@@ -503,95 +506,91 @@ class NodeServer(LabradServer):
     name = 'node %LABRADNODE%'
 
     def __init__(self, nodename, host, port, credential):
+        """Initialize a NodeServer instance.
+
+        Args:
+            nodename (str): The name of this node, e.g. from the LABRADNODE
+                environment variable.
+            host (str): The host where the labrad manager is running.
+            port (int): The port where the labrad manager is running.
+            credential (labrad.auth.Password): Credentials for connecting to the
+                labrad manager.
+        """
         LabradServer.__init__(self)
         self.nodename = nodename
         self.name = 'node %s' % nodename
         self.host = host
         self.port = port
         self.credential = credential
+
+        # Mapping from server name to ServerConfig that describes how to launch
+        # instances of the server.
         self.servers = {}
+
+        # Mapping from instance name to ServerProcess object. Instances are
+        # added to this mapping when they are successfully started and removed
+        # when they send a message indicating that the server process has
+        # stopped.
         self.instances = {}
-        self.initMessages(True)
+
+        # ServerProcess instances that will be notified when new servers connect
```

Comment:
So let me see if I understand correctly: Server A wants to connect to labrad, but requires Server B to start first. Server A can't be a member of `self.instances` because it hasn't started yet, but we need to store a reference to it somewhere so it is informed of Server B successfully connecting. The fact that `self.instances_to_notify` is a `WeakSet` means that if Server A is deleted from `self.instances`, it is also deleted from `self.instances_to_notify`.